src/nvim/regexp_nfa.c

```diff
@@ -6064,10 +6070,17 @@ static int nfa_regmatch(nfa_regprog_T *prog, nfa_state_T *start, regsubs_T *subm
       break;
 
     // Allow interrupting with CTRL-C.
-    fast_breakcheck();
+    line_breakcheck();
```

Comment:
Looks like for unittest `buffer_spec.lua` somehow the queue `loop.uv.pending_queue` was not initialized (`event_init()` or `loop_init()` was not called?) or was left in an undefined state after use.
The queue is not empty but it's `prev` and `next` pointer point to `NULL`.

```c
Program terminated with signal SIGSEGV, Segmentation fault.
#0  0x00007f50d3d21928 in uv__run_pending (loop=0x7f50d3fe3360 <loop>)
    at /home/oni-link/git/neovim/.deps/build/src/libuv/src/unix/core.c:726
726	  QUEUE_SPLIT(&loop->pending_queue, q, &pq);
(gdb) bt
#0  0x00007f50d3d21928 in uv__run_pending (loop=0x7f50d3fe3360 <loop>)
    at /home/oni-link/git/neovim/.deps/build/src/libuv/src/unix/core.c:726
#1  uv_run (loop=0x7f50d3fe3360 <loop>, mode=UV_RUN_NOWAIT) at /home/oni-link/git/neovim/.deps/build/src/libuv/src/unix/core.c:333
#2  0x00007f50d3d1c86c in loop_poll_events (loop=0x7f50d3fe3360 <loop>, ms=0) at /home/oni-link/git/neovim/src/nvim/event/loop.c:49
#3  0x00007f50d3d03139 in os_breakcheck () at /home/oni-link/git/neovim/src/nvim/os/input.c:150
#4  0x00007f50d3c6438f in line_breakcheck () at /home/oni-link/git/neovim/src/nvim/misc1.c:2717
#5  0x00007f50d3bfaeca in nfa_regmatch (prog=0x128bbc0, start=0x128bcd8, submatch=0x7fff66d727a0, m=0x7fff66d72990)
    at /home/oni-link/git/neovim/src/nvim/regexp_nfa.c:6079
#6  0x00007f50d3bfb065 in nfa_regtry (prog=0x128bbc0, col=0, tm=0x0) at /home/oni-link/git/neovim/src/nvim/regexp_nfa.c:6151
#7  0x00007f50d3bfb862 in nfa_regexec_both (line=0x128b6b0 "path_test_file", startcol=0, tm=0x0)
    at /home/oni-link/git/neovim/src/nvim/regexp_nfa.c:6315
#8  0x00007f50d3bfbbf7 in nfa_regexec_nl (rmp=0x7fff66d72d90, line=0x128b6b0 "path_test_file", col=0, line_lbr=false)
    at /home/oni-link/git/neovim/src/nvim/regexp_nfa.c:6467
#9  0x00007f50d3bfbecc in vim_regexec_both (rmp=0x7fff66d72d90, line=0x128b6b0 "path_test_file", col=0, nl=false)
    at /home/oni-link/git/neovim/src/nvim/regexp.c:7059
#10 0x00007f50d3bfc09d in vim_regexec (rmp=0x7fff66d72d90, line=0x128b6b0 "path_test_file", col=0)
    at /home/oni-link/git/neovim/src/nvim/regexp.c:7098
#11 0x00007f50d3bd0f83 in fname_match (rmp=0x7fff66d72d90, name=0x128b6b0 "path_test_file", ignore_case=false)
    at /home/oni-link/git/neovim/src/nvim/buffer.c:1933
#12 0x00007f50d3bd0ee3 in buflist_match (rmp=0x7fff66d72d90, buf=0x128f140, ignore_case=false)
    at /home/oni-link/git/neovim/src/nvim/buffer.c:1914
#13 0x00007f50d3bd09b9 in buflist_findpat (pattern=0x40de8410 "path", pattern_end=0x0, unlisted=0, diffmode=0, curtab_only=0)
    at /home/oni-link/git/neovim/src/nvim/buffer.c:1781
#14 0x00000000004207df in ?? ()
#15 0x0000000000000005 in ?? ()
#16 0x0000000000000000 in ?? ()
```